wallet-new/src/Cardano/Wallet/Server/Plugins.hs

```diff
@@ -123,31 +123,33 @@ legacyWalletBackend :: (HasConfigurations, HasCompileInfo)
                     -> Plugin WalletWebMode
 legacyWalletBackend pm txpConfig WalletBackendParams {..} ntpStatus = pure $ \diffusion -> do
     modifyLoggerName (const "legacyServantBackend") $ do
-      logInfo $ sformat ("Production mode for API: "%build)
-        walletProductionApi
-      logInfo $ sformat ("Transaction submission disabled: "%build)
-        walletTxCreationDisabled
+        logInfo $ sformat ("Production mode for API: "%build)
+            walletProductionApi
+        logInfo $ sformat ("Transaction submission disabled: "%build)
+            walletTxCreationDisabled
 
-      ctx <- view shutdownContext
-      let
-        portCallback :: Word16 -> IO ()
-        portCallback port = usingLoggerName "NodeIPC" $ flip runReaderT ctx $ startNodeJsIPC port
-      walletServeImpl
-        (getApplication diffusion)
-        walletAddress
-        -- Disable TLS if in debug mode.
-        (if isDebugMode walletRunMode then Nothing else walletTLSParams)
-        (Just $ setOnExceptionResponse exceptionHandler defaultSettings)
-        (Just portCallback)
+        ctx <- view shutdownContext
+        let
+            portCallback :: Word16 -> IO ()
+            portCallback port =
+                usingLoggerName "NodeIPC"
+                $ flip runReaderT ctx
+                $ startNodeJsIPC port
+        walletServeImpl
+            (getApplication diffusion)
+            walletAddress
+            -- Disable TLS if in debug mode.
+            (if isDebugMode walletRunMode then Nothing else walletTLSParams)
+            (Just $ setOnExceptionResponse exceptionHandler defaultSettings)
+            (Just portCallback)
   where
     -- Gets the Wai `Application` to run.
     getApplication :: Diffusion WalletWebMode -> WalletWebMode Application
     getApplication diffusion = do
         logInfo "Wallet Web API has STARTED!"
         wsConn <- getWalletWebSockets
         ctx <- V0.walletWebModeContext
-        return
-            $ withMiddleware walletRunMode
+        withMiddleware walletRunMode
```

Comment:
`withMiddleware` is now a monadic function, so we can replace `return . withMiddleware` with `withMiddleware` now.